manifests/plugin/discovery.pp

```diff
@@ -33,15 +33,19 @@
   if $install_images {
     include ::tftp::params
 
-    foreman::remote_file {"${::tftp::params::root}boot/${kernel}":
-      remote_location => "${source}${kernel}",
+    foreman::remote_file {"${::tftp::params::root}boot/${image_name}":
```

Comment:
It might be good to add $tftp_root as a class parameter which defaults to tftp::params::root, so it can be overridden like https://github.com/theforeman/puppet-foreman_proxy/commit/837bfccbf6318d259d861838fbfec9a9d20d7e03?  (Include tftp::params in foreman::plugin::discovery::params.)